packages/cli/src/handlers/dbt/getWarehouseClient.ts

```diff
@@ -360,28 +383,84 @@ export default async function getWarehouseClient(
             console.error(`\n✓ Successfully authenticated with Databricks\n`);
         }
 
-        // Check if we should use cached client (e.g., for auth methods requiring user interaction)
-        const cacheKey = getWarehouseClientCacheKey(credentials);
-
-        if (warehouseClientCache.has(cacheKey)) {
-            GlobalState.debug(
-                `> Reusing cached warehouse client (${credentials.type})`,
+        // Handle Snowflake External Browser authentication
+        // We authenticate via browser, create a PAT, and use that for subsequent operations
+        if (
+            credentials.type === WarehouseTypes.SNOWFLAKE &&
+            credentials.authenticationType ===
+                SnowflakeAuthenticationType.EXTERNAL_BROWSER &&
+            !options.skipPatCreation
```

Comment:
we can only generate 15 tokens per user, so we need to optimize them: 

if user is updating an existing project or preview, we don't generate a new PAT, since credentials are not overwritten (for now)